### DISCUSSION TITLE
Deploy dialog: pick environment from a list

### DISCUSSION THREAD

**Discussion by rigrig** [woodpecker-ci/woodpecker#2171] [Request a Feature]
### Clear and concise description of the problem

I would like to pick a deployment environment from a list when pressing the deploy button, instead of having to type it in.

### Suggested solution

* In the "Deploy" dialog, allow picking "Target deployment environment" from a list of all environments listed in the pipeline

### Alternative

* On the repository settings page, add a "Deployment environments" input, for a "Comma-separated list of deployment environments"
* In the "Deploy" dialog, allow picking an option from the environments listed in the settings. 

### Additional context

I only have a single environment anyway, so needing to remember/look up what I called it and type that in feels kind of pointless.

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by 6543** [comment]
I think the "Alternative" is easyer to do, as we have the info already in DB, we just need an function in the "datastore" module and a rest endpoint backend whise ...

**Comment by rigrig** [comment]
Thanks, I was thinking too much about the implementation instead of just the desired result when I opened the issue, edited now.
